src/textalloc/__init__.py

```diff
@@ -66,13 +66,15 @@ def allocate(
     linewidth: float = 1,
```

Comment:
@ckjellson 
I see your point for the ordering that could be done in advance

Most of the time you just want to optimize and plot an unordered set of text labels.
However in my case the order of the textboxes already has a meaning. For instance, I have a list of M+N+K texts that i want to allocate together, and then store the results at different places so that in my GUI the user can disable the N texts in the middle or change the color of the K last texts. It would make the code less readable to add this argsort mechanism before and after the call to textalloc 

Reordering the texts only makes sense for the optimization, so imo it shouldn't be managed externally by the user, who uses text_alloc specifically not to worry about the optim! ;)

Reply:
By the way, i need to check that the output order is correctly kept. It thought it will be ok with the overlapping_boxes_inds

Reply:
I see. Then I think the argsort_priority should be the input parameter named for example label_priorities and you can then pass any priority order you want. Otherwise I can imagine infinitely many parameters (prioritize_longest_texts, prioritize_shortest_texts, prioritize_texts_with_most_as, prioritize_texts_with_most_bs...). Also I think it would make sense to move the random seed to the place where you are now doing the prioritizing, so that:

__init__.py
...
label_priorities = np.array(label_priorities)
for i in range(len(label_priorities)): assert i in label_priorities
assert len(label_priorities)==len(np.unique(label_priorities))
...
**remove seed logic**

non_overlapping_boxes.py:110:
if label_priorities is not None:
    argsort_priority = label_priorities
elif seed>0:
    argsort_priority = np.arange(len(original_boxes))
    np.random.seed(seed)
    np.random.shuffle(argsort_priority)
else:
    argsort_priority = list(range(len(original_boxes)))

If you don't do it, I will also change so that the text_objects and line_objects lists match the result_text_xy and result_lines lists by adding None for texts that were not plotted.

Reply:
The practical reason why i had to introduce this priority is that, because of the greedy nature of this allocation, I often get the case where swapping the order of two labels will turn a successful optim into a failed one... This is more a priority to ensure that all texts will be correctly allocated, rather than a priority to make some texts better allocated than others. And in this case, the only priority that seems to help is putting large texts first and then fill the gaps with the small ones, thus i'm not sure about the need of priorities like "prioritize_shortest_texts"

but you're right, it's much cleaner to wrap all this into a single priority strategy.

What about taking an union of these 3 types as strategy input?
- int : seed
- str: Strategy name (among the available list)
- Callable[[str, str], bool]: sorting method to compare two text strings

Reply:
Okay, that sounds reasonable, but I could also imagine cases when you would actually want some priority order of which labels to put into the plot and which ones should be left out if they do not fit. I agree it might have been a bad example 😄
Union of strategies sounds good!

Reply:
Great! the `callable` type will handle all your weird priority wishes ;)

Reply:
@ckjellson 
done

Reply:
By the way @ckjellson, how is the new release on PyPi triggered ?